### ISSUE TITLE
Swift: Properties set on outlets in `didSet {}` get overwritten

### ISSUE THREAD

**Issue by aceontech** [gk-brown/MarkupKit#78]
This is related to using the `didSet` property observer in Swift in combination with outlets managed by MarkupKit: It seems that properties are getting overwritten when set in `didSet`. For example, consider the following markup:

```xml
<LMRowView>
  <UILabel id="label1" text="LABEL1"/>
  <UILabel id="label2" text="LABEL2"/>
</LMRowView>
```

and the following Swift code: 

```Swift
@IBOutlet weak var label1:UILabel? {
  didSet {
    label1.text = "new text"
  }
}
```

At runtime, `label1`'s text property will be displayed as `LABEL1` instead of the expected `new text`.

I think this is because `LMViewBuilder` applies properties and action handlers *after* setting the outlet on the owner. Swift immediately calls the `didSet` property observer while LMViewBuilder hasn't yet completed setting the outlet's properties. See [LMViewBuilder line 533](https://github.com/gk-brown/MarkupKit/blob/master/MarkupKit-iOS/MarkupKit/LMViewBuilder.m#L533) and beyond.

Shouldn't the outlet be set *after* properties and actions have been applied, i.e. right before the view gets pushed to the view stack? 

In addition to being more semantically correct, I think this is also how Nibs and Storyboards work internally.

**Comment by aceontech** [comment]
I should point out this bug doesn't occur consistently and seems to be timing-sensitive. 

In my test case, I was loading the markup lazily in `UIView.updateConstraints()`, which delays the parsing (as opposed to loading in the initializer or `viewDidLoad` of a `UIViewController`).

**Comment by gk-brown** [comment]
I'll give this some more thought, but I think the existing behavior is valid. If I declare a member variable named `label` as follows:

    @IBOutlet var label: UILabel! {
        didSet {
            label.text = "xyz"
        }
    }

then this markup:

    <UILabel id="label" text="abc"/>

is analogous to this Swift code:

    label = UILabel()
    label.text = "abc"

The label instance is first allocated and assigned to the `label` property, and then the label's properties are set. In either case, the results would be the same: the label would read "abc".

What you are describing is more akin to this code:

    label = UILabel()
    label.text = "abc"

   self.label = label

Both are valid, but the first is slightly simpler and easier to explain/document.

Additionally, this is to some extent an implementation detail. `LMViewBuilder` guarantees that all properties have been set and all outlets populated by the time `viewWithName:owner:root:` returns. It doesn't attempt to make any guarantees about the order in which those operations will occur.

So I believe your current approach is correct. Views associated with view controllers should generally be loaded in the controller's `loadView` method, passing the controller as `owner` and `nil` for `root`, and views whose root element is already defined (such as a custom table view cell) should be loaded in the initializer, passing the root view as both `owner` and `root`.

**Comment by gk-brown** [comment]
Since there haven't been any recent comments on this issue, I'm going to close it. We can revisit later if needed.

**Comment by gk-brown** [comment]
The requested behavior has been implemented in the development branch for 2.8. If possible, please test and verify - thanks.
